sign/sign.go

```diff
@@ -110,7 +115,7 @@ func MappedTransaction(tx map[string]interface{}) (*Transaction, error) {
 	if err := s.checkRequiredFields(); err != nil {
 		return nil, err
 	}
-	s.populateHash()
+	s.PopulateHash()
```

Comment:
Are we not using the writing the custom JSON marshal/unmarshal? Under no circumstances should populate hash be a public method; this is a footgun.

Reply:
I wanted to try to get this merged today, so I created a separate task to track that <https://linear.app/arguslabs/issue/WORLD-1224/fix-abstraction-leakage-in-cardinalserverhandlertxgo> [](edited-with-Axolo)

Reply:
I have been working on it in a branch. [](edited-with-Axolo)